monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppManager.java

```diff
@@ -126,7 +103,8 @@ public void deleteApp(String id) {
         if (app != null) {
             appTestDbConnection.deleteTestsForApp(id);
             dbConnection.delete(app);
-            dbConnection.deleteGroup(dbConnection.getGroupById(app.getGroup().getName()));
+            if (app.getGroup().getApps().size() == 1)
```

Comment:
This check is done  to delete the group if the group's last app is deleted

Reply:
why? a group should exist without apps (e.g. remove the old apps from a group after they have to a different server and then create new apps or something like that).